streams/src/test/java/org/apache/kafka/streams/integration/EosIntegrationTest.java

```diff
@@ -668,6 +669,14 @@ public void close() { }
         return streams;
     }
 
+    private void startAndWaitForRunningState(final KafkaStreams streams) throws Exception {
```

Comment:
Maybe worth lifting to TestUtils since this pattern is showing up a lot?

---

Minor nit: if you use a CountDownLatch then you don't need to poll and thus can respond a bit more quickly to completion. I think the default resolution on wait is 100ms so it's not a huge difference, but it also seems a bit simpler to do a timed wait on CDL? Your choice.

Reply:
Ack.

Reply:
Ack